tests/bluetooth/host/id/bt_id_read_public_addr/src/main.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/fff.h>
+#include <zephyr/bluetooth/hci.h>
+#include <host/hci_core.h>
+#include <host/id.h>
+#include "mocks/hci_core.h"
+#include "mocks/hci_core_expects.h"
+#include "mocks/net_buf.h"
+#include "mocks/net_buf_expects.h"
+#include "testing_common_defs.h"
+
+DEFINE_FFF_GLOBALS;
+
+/* Hold data representing HCI command response for command BT_HCI_OP_READ_BD_ADDR */
+static struct net_buf hci_cmd_rsp;
+
+/* Hold data representing response data for HCI command BT_HCI_OP_READ_BD_ADDR */
+static struct bt_hci_rp_read_bd_addr hci_rp_read_bd_addr;
+
+static void tc_setup(void *f)
+{
+	memset(&bt_dev, 0x00, sizeof(struct bt_dev));
+	memset(&hci_cmd_rsp, 0x00, sizeof(struct net_buf));
+	memset(&hci_rp_read_bd_addr, 0x00, sizeof(struct bt_hci_rp_read_bd_addr));
+	NET_BUF_FFF_FAKES_LIST(RESET_FAKE);
+	HCI_CORE_FFF_FAKES_LIST(RESET_FAKE);
+}
+
+ZTEST_SUITE(bt_id_read_public_addr, NULL, NULL, tc_setup, NULL, NULL);
+
+/*
+ *  Test reading controller public address through bt_hci_cmd_send_sync(), but
+ *  operation fails and a non-success error code is returned.
+ *
+ *  Constraints:
+ *   - A valid address reference is passed to bt_id_read_public_addr()
+ *   - bt_hci_cmd_send_sync() returns a non-zero error code
+ *
+ *  Expected behaviour:
+ *   - Execution stops and a zero return value is returned which represents failure
+ */
+ZTEST(bt_id_read_public_addr, test_bt_hci_cmd_send_sync_returns_err)
+{
+	int err;
+	bt_addr_le_t addr;
+
+	bt_hci_cmd_send_sync_fake.return_val = !0;
```

Comment:
```suggestion
	bt_hci_cmd_send_sync_fake.return_val = 1;
```

Reply:
Ok, but it was intended to be read as `NOT 0`